openforcefield/tests/test_molecule.py

```diff
@@ -529,6 +529,109 @@ def test_name(self):
+                                       bond_order_matching=False, atom_stereochemistry_matching=False,
+                                       bond_stereochemistry_matching=False)[0] is True
+        # test hill formula passes but isomorphic fails
+        mol1 = Molecule.from_smiles('Fc1ccc(F)cc1')
+        mol2 = Molecule.from_smiles('Fc1ccccc1F')
+        assert mol1.is_isomorphic_with(mol2) is False
+        assert mol2.is_isomorphic_with(mol1) is False
+
+    def test_remap(self):
+        """Test the remap function which should return a new molecule in the requested ordering"""
+        # the order here is CCO
+        ethanol = Molecule.from_file(get_data_file_path('molecules/ethanol.mol2'))
+        # get ethanol in reverse order OCC
+        ethanol_reverse = Molecule.from_smiles('OCC')
+        # get the mapping between the molecules
+        mapping = Molecule.are_isomorphic(ethanol, ethanol_reverse, True)[1]
+        ethanol.add_bond_charge_virtual_site([0, 1], 0.3 * unit.angstrom)
+        # make sure that molecules with virtual sites raises an error
+        with pytest.raises(Exception):
+            remapped = ethanol.remap(mapping, current_to_new=True)
+
+        # remake with no virtual site and remap to match the reversed ordering
+        ethanol = Molecule.from_file(get_data_file_path('molecules/ethanol.mol2'))
+
+        new_ethanol = ethanol.remap(mapping, current_to_new=True)
+
+        def assert_molecules_match_after_remap(mol1, mol2):
+            """Check all of the attributes in a molecule match after being remapped"""
+            for atoms in zip(mol1.atoms, mol2.atoms):
+                assert atoms[0].to_dict() == atoms[1].to_dict()
+            # bonds will not be in the same order in the molecule and the atom1 and atom2 indecies could be out of order
+            remaped_bonds = [bond.to_dict() for bond in mol2.bonds]
+            for bond in mol1.bonds:
+                assert bond.to_dict() in remaped_bonds
+            assert mol1.n_bonds == mol2.n_bonds
+            assert mol1.n_angles == mol2.n_angles
+            assert mol1.n_propers == mol2.n_propers
+            assert mol1.n_impropers == mol2.n_impropers
+            assert mol1.total_charge == mol2.total_charge
+
+        # check all of the properties match as well, torsions and impropers will be in a different order
+        # due to the bonds being out of order
+        assert_molecules_match_after_remap(new_ethanol, ethanol_reverse)
+
+        # catch mappings that are the wrong size
+        too_small_mapping = {0: 1}
+        with pytest.raises(Exception):
+            new_ethanol = ethanol.remap(too_small_mapping, current_to_new=True)
+
+        wrong_index_mapping = {(i + 10, new_id) for i, new_id in enumerate(mapping.values())}
+        with pytest.raises(Exception):
+            new_ethanol = ethanol.remap(wrong_index_mapping, current_to_new=True)
```

Comment:
It's awesome that you're checking this. Maybe break these checks out into a separate test?

Reply:
Thanks, I have separated them out.